geopaparazzilibrary/src/eu/geopaparazzi/library/gps/GpsAvgMeasurements.java

```diff
@@ -0,0 +1,232 @@
+	public synchronized double getLatitude() {
+		return averageLat;
+	}
+
+	/** @return weighted mean of all longitudes in list, or 0 if list is empty. */
+	public synchronized double getLongitude() {
+		return averageLon;
+	}
+
+	/** @return arithmetic mean of all errors in list, or 0 if list is empty. */
+	public synchronized float getAccuracy() {
+		return averageAccuracy;
+	}
+
+	/** @return weighted mean of all altitudes in list, or 0 if list is empty. */
+	public synchronized double getAltitude() {
+		return averageAlt;
+	}
+
+	/**
+	 * @param index
+	 *            in list, 0 for first item, size()-1 for last item
+	 * @return stored latitude from list.
+	 */
+	public synchronized double getLatitude(int index) {
+		return locations.get(index).getLatitude();
+	}
+
+	/**
+	 * @param index
+	 *            in list, 0 for first item, size()-1 for last item
+	 * @return stored longitude from list.
+	 */
+	public synchronized double getLongitude(int index) {
+		return locations.get(index).getLongitude();
+	}
+
+	/**
+	 * @param index
+	 *            in list, 0 for first item, size()-1 for last item
+	 * @return stored error from list.
+	 */
+	public synchronized float getAccuracy(int index) {
+		return locations.get(index).getAccuracy();
+	}
+
+	/**
+	 * @param index
+	 *            in list, 0 for first item, size()-1 for last item
+	 * @return stored altitude from list.
+	 */
+	public synchronized double getAltitude(int index) {
+		return locations.get(index).getAltitude();
+	}
+
+	/**
+	 * @param index
+	 *            in list, 0 for first item, size()-1 for last item
+	 * @return date of store index in list.
+	 */
+	public synchronized Date getTime(int index) {
+		return new Date(locations.get(index).getTime());
+	}
+
+	@Override
+	public synchronized String toString() {
+
+		StringBuffer sb = new StringBuffer();
+
+		sb.append(getClass().getSimpleName()).append(": (");
+		sb.append("avg lat=").append(getLatitude());
+		sb.append(" lon=").append(getLongitude());
+		sb.append(" alt=").append(getAltitude());
+		sb.append(" acc=").append(getAccuracy()).append(" ");
+		for (int i = 0; i < size(); i++) {
+			sb.append("[").append(getLongitude(i)).append(",");
+			sb.append(getLatitude(i)).append(",");
+			sb.append(getAltitude(i)).append(",");
+			sb.append(getAccuracy(i)).append("]");
+		}
+		sb.append(")");
+
+		return sb.toString();
+	}
+
+	/** Calculates distance between two gps lat/lon pairs */
+	private double distance(double lat1, double lng1, double lat2, double lng2) {
+		double earthRadius = 3958.75;
+		double dLat = Math.toRadians(lat2 - lat1);
+		double dLng = Math.toRadians(lng2 - lng1);
+		double a = Math.sin(dLat / 2) * Math.sin(dLat / 2) + Math.cos(Math.toRadians(lat1))
+				* Math.cos(Math.toRadians(lat2)) * Math.sin(dLng / 2) * Math.sin(dLng / 2);
+		double c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
+		double dist = earthRadius * c;
+
+		int meterConversion = 1609;
+		return (dist * meterConversion);
+	}
+
```

Comment:
Yes, I thought we had it, but, as I was just porting it over unchanged, I was avoiding modifying this class for now. We certainly can remove it if it is the same. Right now, I don't call this function, BUT with some of EI's ideas, we might want to.